api/src/test/java/org/openmrs/api/OrderServiceTest.java

```diff
@@ -3794,12 +3800,139 @@ public void saveOrderGroup_shouldFailValidationIfAnyOrdersFailValidation() {
+		assertEquals(4,orderGroupAttributeTypes.size());
+	}
+	
+	@Test
+	public void getOrderGroupAttributeType_shouldReturnNullIfNonExistingIdIsProvided(){
+		executeDataSet(ORDER_GROUP_ATTRIBUTES);
+		assertNull(Context.getOrderService().getOrderGroupAttributeTypeById(10));
+	}
+
+	@Test 
+	public void getOrderGroupAttributeType_shouldReturnOrderGroupAttributeTypeGivenId(){
+		executeDataSet(ORDER_GROUP_ATTRIBUTES);
+		final Integer ID = 2;
+		final String UUID2="9cf1bbe6-d18e-11ea-87d0-0242ac130003";
+		OrderGroupAttributeType orderGroupAttributeType = orderService.getOrderGroupAttributeTypeById(ID);
+		assertEquals(orderService.getOrderGroupAttributeTypeByUuid(UUID2),orderGroupAttributeType);
+		assertEquals(orderGroupAttributeType.getUuid(),Context.getOrderService().getOrderGroupAttributeTypeByUuid(UUID2).getUuid());
+		assertEquals(ID,Context.getOrderService().getOrderGroupAttributeTypeByUuid(UUID2).getId());
+	}
+
+	@Test
+	public void getOrderGroupAttributeTypeByUuid_shouldReturnOrderGroupAttributeTypeByUuid(){
+		executeDataSet(ORDER_GROUP_ATTRIBUTES);
+		OrderGroupAttributeType orderGroupAttributeType = orderService.getOrderGroupAttributeTypeByUuid("9cf1bce0-d18e-11ea-87d0-0242ac130003");
+		assertEquals("Bacteriology",orderGroupAttributeType.getName());
+	}
+
+	@Test
+	public void saveOrderGroupAttributeType_shouldSaveOrderGroupAttributeTypeGivenOrderGroupAttributeType() throws ParseException {
+		executeDataSet(ORDER_GROUP_ATTRIBUTES);
+		int initialGroupOrderAttributeTypeCount = Context.getOrderService().getOrderGroupAttributeTypes().size();
+		OrderGroupAttributeType orderGroupAttributeType = new OrderGroupAttributeType();
+		orderGroupAttributeType.setName("Surgery");
+		orderGroupAttributeType.setDatatypeClassname(FreeTextDatatype.class.getName());
+		Context.getOrderService().saveOrderGroupAttributeType(orderGroupAttributeType);
+		assertNotNull(orderGroupAttributeType.getId());
+		assertEquals(initialGroupOrderAttributeTypeCount+1,Context.getOrderService().getOrderGroupAttributeTypes().size());
+	}
+		
+	@Test
+	public void saveOrderGroupAttributeType_shouldEditAnExistingOrderGroupAttributeType(){
+		executeDataSet(ORDER_GROUP_ATTRIBUTES);
+		final String UUID4 ="9cf1bdb2-d18e-11ea-87d0-0242ac130003";
+		final String name ="ECG";
+		//Check for values in the database
+		OrderGroupAttributeType orderGroupAttributeType=Context.getOrderService().getOrderGroupAttributeTypeById(4);
+		assertEquals(UUID4,orderGroupAttributeType.getUuid());
+		assertEquals(name,orderGroupAttributeType.getName());
+		assertEquals("Testing unretire",orderGroupAttributeType.getRetireReason());
+        //edit existing values in the database
+		orderGroupAttributeType.setRetireReason("Change Order To Laparascopy");
+	    orderGroupAttributeType.setName("Laparascopy");
+		orderService.saveOrderGroupAttributeType(orderGroupAttributeType);
+        //confirm new values are persisted
+		assertEquals("Laparascopy",orderGroupAttributeType.getName());
```

Comment:
What value does assertEquals and then assertNotEquals name add?

Reply:
I added these to confirm that the method has updated the name and other fields in the database . That's why it compares the previous name "ECG" with the new name "Laparascopy" . But i will remove it since it's unnecessary.

Reply:
I added these to confirm that the method has updated the name and other fields in the database . That's why it compares the previous name "ECG" with the new name "Laparascopy" . But i will remove it since it's unnecessary.

Reply:
What you are doing is similar to:
assertEquals(name, 'tendo');
assertNotEquals(name, 'martin');

Reply:
Ooh ok sure, am rectifying this as well.